core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -969,9 +988,26 @@ wasi_poll_oneoff(wasm_exec_env_t exec_env, const wasi_subscription_t *in,
         || !validate_native_addr(nevents_app, sizeof(uint32)))
         return (wasi_errno_t)-1;
 
-    err = wasmtime_ssp_poll_oneoff(curfds, in, out, nsubscriptions, &nevents);
-    if (err)
-        return err;
+    int elapsed = 0;
+    /* in seconds */
+    const int timeout = get_timeout_for_poll_oneoff(in), time_quant = 1;
+    __wasi_subscription_t in_copy[nsubscriptions];
+    memcpy(in_copy, in, nsubscriptions * sizeof(wasi_subscription_t));
```

Comment:
Please use bh_memcpy_s to improve safety

Reply:
done